src/modules/ffmpeg/producer/av_producer.cpp

```diff
@@ -104,8 +104,12 @@ struct Decoder
 
         FF(av_opt_set_int(ctx.get(), "refcounted_frames", 1, 0));
 
-        // TODO (fix): Remove limit.
-        FF(av_opt_set_int(ctx.get(), "threads", env::properties().get(L"configuration.ffmpeg.producer.threads", 4), 0));
+        int defaultNumThreads =
+            codec->capabilities & AV_CODEC_CAP_AUTO_THREADS
+                ? 0 : codec->capabilities & (AV_CODEC_CAP_FRAME_THREADS | AV_CODEC_CAP_SLICE_THREADS)
```

Comment:
With slice threads it makes sense to limit it to e.g 8 threads... since it doesn't scale very well.

Reply:
So you are proposing min<int>(8, concurrency/2) when slice threads, concurrency/2 otherwise?

Reply:
Done!